### PR TITLE
followRedirects option and more strict type checking

### PR DESCRIPTION
**Pull request by tunnckoCore**
- add followRedirects option (enabled by default, as it was)
- more strict type checking
- add gitignore

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,10 +16,12 @@ function simpleGet (opts, cb) {
     cb(new Error('too many redirects'))
     return
   }
-  if (!opts.maxRedirects) opts.maxRedirects = 10
+
+  opts.followRedirects = typeof opts.followRedirects === 'boolean' ? opts.followRedirects : true
+  opts.maxRedirects = typeof opts.maxRedirects === 'number' ? opts.maxRedirects : 10
+  opts.headers = typeof opts.headers === 'object' ? opts.headers : {}
```
Comment (feross) [inline index.js]: This would accept a value of `null` since `typeof null === 'object'`.

File: index.js [hunk 2]
```diff
@@ -16,10 +16,12 @@ function simpleGet (opts, cb) {
     cb(new Error('too many redirects'))
     return
   }
-  if (!opts.maxRedirects) opts.maxRedirects = 10
+
+  opts.followRedirects = typeof opts.followRedirects === 'boolean' ? opts.followRedirects : true
```
Comment (feross) [inline index.js]: We don't need a new option for this. We can just use the existing `maxRedirects` option. If `maxRedirects === 0` then no redirects should be followed.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (feross) [inline .gitignore]: You can add this to your global .gitignore instead of putting it in every repo you send a PR to.

**Comment by feross** [comment]
Comments inline. I'll accept after you've addressed the comments.

File: .gitignore (hunk 3 above)
Comment (tunnckoCore) [inline .gitignore]: i have it https://github.com/tunnckoCore/dotfiles

Comment (tunnckoCore) [inline .gitignore]: didn't work, which really was strange, yea. that's why i added it. just delete it after merge?

**Comment by feross** [comment]
Also, this is missing a test. Can you add a test that `maxRedirects === 0` causes no redirects to be followed.

File: .gitignore (hunk 3 above)
Comment (feross) [inline .gitignore]: Since you'll need to update this PR anyway, can you just remove it?

**Comment by tunnckoCore** [comment]
Mmm. Cant figure it out. example? I just tried few tests, some of them was

- when server redirects, but `followRedirects:false`
- when server redirects and `simple-get` have `followRedirects:false` and `maxRedirects:0`

but coverage remains the same

**Comment by feross** [comment]
What is the intended behavior when `followRedirects === false`? I don't understand what you're trying to make this do. Adding some tests with this PR would help.

In the meantime, I've adopted a few of the suggestions in here, but not the `followRedirects` option.

**Pull request closed without merging**